.devcontainer/Dockerfile

```diff
@@ -58,6 +51,10 @@ RUN \
     rm -rf /tmp/* && \
     rm -rf /var/lib/apt/lists/*
 
+# Install azure-cli
+RUN pip install setuptools==57.0.0 \
```

Comment:
>setuptools==57.0.0 [](http://example.com/codeflow?start=16&length=18)

i believe that this dependency is managed internally by the az cli?

Reply:
It is true that az cli installs `setuptools==52.0.0` [requirements.txt](https://github.com/Azure/azure-cli/blob/162be08eefaa1d64974a07b6179babe4c5b2830a/requirements.txt#L2). However, we still need to make sure that the version of setuptools is <58.0.0 because of the following:
- The installation of python3.9 includes a newer version of setuptools (incompatible with az cli 2.25.0);
- az cli  has a dependency on `pygithub==1.38`, which may get installed before `setuptools` is downgraded to 52.0.0 (please note that there's no way to specify the order of installation with pip);
- this version of `pygithub==1.38` needs setuptools <58.0.0 because of a deprecated featured (see [code](https://github.com/PyGithub/PyGithub/blob/v1.38/setup.py#L104)).

Ideally, we should be using newer versions of az cli, but `azure-cli-core` is pinned to 2.25.0 (see [code](https://github.com/Azure/iotedgedev/blob/55a4883a5006e85ccb3d1f33195feae803b7dc78/requirements.txt#L10)).

Reply:
I'd recommend to use a newer version of az cli instead of trying to account for deprecated lib versions.

Reply:
Yes, that would be ideal, but as mentioned, `iotedgedev` pins `azure-cli-core` to 2.25.0 (see [code](https://github.com/Azure/iotedgedev/blob/55a4883a5006e85ccb3d1f33195feae803b7dc78/requirements.txt#L10)), which means that  the version of az-cli installed via pip needs to be the same.